sklearn/ensemble/_forest.py

```diff
@@ -1413,6 +1422,9 @@ class RandomForestRegressor(ForestRegressor):
         valid partition of the node samples is found, even if it requires to
         effectively inspect more than ``max_features`` features.
 
+        .. deprecated:: 1.0
+            Option "auto" will be changed to `sqrt(n_features)` in v1.2.
```

Comment:
I find this confusing. `max_features="auto"` will still be valid and the default in v1.2. I am not sure we need the `deprecated:: 1.0` directive at all here. I think the note in the changelog and the warning are enough. But if we want to keep it, let rephrase to:

```suggestion
            Option "auto" will behave as `sqrt(n_features)` in v1.2.
```

Reply:
With or without `.. deprecated` directive, I'm fine with both.